docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,214 @@
+    # For the above endpoint - the two calls are defined, one uses GET method - the custom_script will be executed
+    # there, second uses the POST method and then the another_custom_script will be called;
+    # Currently only script are available for calls;
+
+    # After the creation of the endpoint, add them to custom_socket:
+    custom_socket.add_endpoint(my_endpoint)
+
+    # define dependency now;
+    # using a new script - defining new source code;
+    custom_socket.add_dependency(
+        ScriptDependency(
+            Script(
+                label='custom_script',
+                runtime_name=RuntimeChoices.PYTHON_V5_0,
+                source='print("custom_script")'
+            )
+        )
+    )
+    # using a existing script:
+    another_custom_script = Script.please.get(id=2)
+    custom_socket.add_dependency(
+        ScriptDependency(
+            another_custom_script
+        )
+    )
+
+    # now it is time to publish custom_socket;
+    custom_socket.publish()  # this will do an api call and will create script;
+
+Some time is needed to setup the environment for this custom socket.
+There is possibility to check the custom socket status::
+
+    print(custom_socket.status)
+    # and
+    print(custom_socket.status_info)
+
+    # to reload object (read it again from syncano api) use:
+    custom_socket.reload()
+
+
+
+Updating the custom socket
+--------------------------
+
+To update custom socket, use::
+
+    custom_socket = CustomSocket.please.get(name='my_custom_socket')
+
+    custom_socket.remove_endpoint(endpoint_name='my_endpoint')
+    custom_socket.remove_dependency(dependency_name='custom_script')
+
+    # or add new:
+
+    custom_socket.add_endpoint(new_endpoint)  # see above code for endpoint examples;
+    custom_socket.add_dependency(new_dependency)  # see above code for dependency examples;
+
+    custom_socket.update()
+
+
+Running the custom socket
+-------------------------
+
+To run custom socket use::
+
+    # this will run the my_endpoint - and call the custom_script (method is GET);
+    result = custom_socket.run(method='GET', endpoint_name='my_endpoint')
+
+
+Read all endpoints
+------------------
+
+To get the all defined endpoints in custom socket run::
+
+    endpoints = custom_socket.get_endpoints()
+
+    for endpoint in endpoints:
+        print(endpoint.name)
+        print(endpoint.calls)
+
+To run particular endpoint::
+
+    endpoint.run(method='GET')
+    # or:
+    endpoint.run(method='POST', data={'name': 'test_name'})
+
+The data will be passed to the api call in the request body.
+
+Custom sockets endpoints
+------------------------
+
+Each custom socket is created from at least one endpoint. The endpoint is characterized by name and
+defined calls. Calls is characterized by name and methods. The name is a identification for dependency, eg.
+if it's equal to 'my_script' - the Script with label 'my_script' will be used (if exist and the source match),
```

Comment:
not exactly.
the Script Endpoint with name 'my_script' will be used (if it exists and Script source and runtime matches),